src/Sylius/Bundle/CoreBundle/Behat/MetadataContext.php

```diff
@@ -0,0 +1,396 @@
+        }
+
+        $em = $this->getEntityManager();
+        $em->persist($metadata);
+        $em->flush();
+    }
+
+    /**
+     * @Given product :productName has the following page metadata:
+     */
+    public function productHasTheFollowingPageMetadata($productName, TableNode $table)
+    {
+        /** @var ProductInterface $product */
+        $product = $this->getRepository('product')->findOneBy(['name' => $productName]);
+
+        $this->thereIsTheFollowingMetadata($product->getMetadataIdentifier(), $table);
+    }
+
+    /**
+     * @Then I should see :title as page title
+     */
+    public function iShouldSeeAsPageTitle($title)
+    {
+        $this->assertSession()->elementTextContains('css', 'title', $title);
+    }
+
+    /**
+     * @Then the page keywords should contain :keyword
+     */
+    public function thePageKeywordsShouldContain($keyword)
+    {
+        $this->assertSession()->elementExists('css', sprintf('meta[name="keywords"][content*="%s"]', $keyword));
+    }
+
+    /**
+     * @Then there should be Twitter summary card metadata on this page
+     */
+    public function thereShouldBeTwitterSummaryCardMetadataOnThisPage()
+    {
+        $this->assertSession()->elementExists('css', 'meta[name="twitter:card"][content="summary"]');
+    }
+
+    /**
+     * @Then Twitter site should be :site
+     */
+    public function twitterSiteShouldBe($site)
+    {
+        $this->assertSession()->elementExists('css', sprintf('meta[name="twitter:site"][content="%s"]', $site));
+    }
+
+    /**
+     * @Then Twitter image should be :image
+     */
+    public function twitterImageShouldBe($image)
+    {
+        $this->assertSession()->elementExists('css', sprintf('meta[name="twitter:image"][content="%s"]', $image));
+    }
+
+    /**
+     * @When /I deselect "([^"]+)"/
+     */
+    public function iDeselectSelectField($fieldName)
+    {
+        $this->selectOption($fieldName, "");
+    }
+
+    /**
+     * @param ElementInterface $element
+     * @param string $regexp
+     *
+     * @throws \Exception If assertion failed
+     */
+    private function assertItIsMetadataCustomizationPage(ElementInterface $element, $regexp)
+    {
+        if ($this->isItMetadataCustomizationPage($element, $regexp)) {
+            return;
+        }
+
+        throw new \Exception(sprintf("It is not metadata customziation page (regexp: %s)", $regexp));
+    }
+
+    /**
+     * @param ElementInterface $element
+     * @param string $regexp
+     *
+     * @return bool
+     */
+    private function isItMetadataCustomizationPage(ElementInterface $element, $regexp)
+    {
+        $header = $element->find('css', '.page-header h1');
+
+        if (null === $header) {
+            return false;
+        }
+
+        if (false === strpos($header->getText(), "Customizing metadata")) {
+            return false;
+        }
+
+        if (!preg_match($regexp, $header->getText())) {
+            return false;
+        }
+
+        return true;
```

Comment:
I don't think it will bring more clarity to this method, there are already two other `if` statements in a row. Anyway, if I would like to add another requirement to be checked after these ones, I'd have to undo the proposed change.